registrations/views.py

```diff
@@ -137,3 +156,109 @@ def get(self, request, *args, **kwargs):
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request):
+        is_client = IdentityStoreApiClient(
+            api_url=settings.IDENTITY_STORE_URL,
+            auth_token=settings.IDENTITY_STORE_TOKEN
+        )
+        # load the users sources - posting users should only have one source
+        source = Source.objects.get(user=self.request.user)
+        serializer = ThirdPartyRegistrationSerializer(data=request.data)
+        if serializer.is_valid():
+            mom_msisdn = serializer.validated_data['mom_msisdn']
+            hcw_msisdn = serializer.validated_data['hcw_msisdn']
+            lang_code = serializer.validated_data['mom_lang']
+            lang_code = transform_language_code(lang_code)
+            if mom_msisdn != hcw_msisdn:
+                # Get or create HCW Identity
+                result = is_client.get_identity_by_address(
+                    'msisdn', hcw_msisdn)
+                if 'results' in result and not result['results']:
+                    identity = {
+                        'details': {
+                            'default_addr_type': 'msisdn',
+                            'addresses': {
+                                'msisdn': {
+                                    hcw_msisdn: {'default': True}
+                                }
+                            }
+                        }
+                    }
+                    hcw_identity = is_client.create_identity(identity)
+                else:
+                    hcw_identity = result['results'][0]
+            else:
+                hcw_identity = None
+
+            id_type = serializer.validated_data['mom_id_type']
+            if hcw_identity is not None:
+                operator = hcw_identity['id']
+                device = hcw_msisdn
+            else:
+                operator = None
+                device = mom_msisdn
+            # Get or create Mom Identity
+            result = is_client.get_identity_by_address(
+                'msisdn', mom_msisdn)
+            if 'results' in result and not result['results']:
+                identity = {
+                    'details': {
+                        'default_addr_type': 'msisdn',
+                        'addresses': {
+                            'msisdn': {
+                                mom_msisdn: {'default': True}
+                            }
+                        }
+                    },
+                    'operator_id': operator,
+                    'lang_code': lang_code,
+                    'id_type': id_type,
+                    'mom_dob': serializer.validated_data['mom_dob'],
+                    'edd': serializer.validated_data['mom_edd'],
+                    'faccode': serializer.validated_data['clinic_code'],
+                    'consent': serializer.validated_data['consent'],
+                    'mha': serializer.validated_data['mha'],
+                    'swt': serializer.validated_data['swt'],
```

Comment:
We set two other fields in the js apps: `last_mc_reg_on` and `source`.
`source` should keep track of which app created the identity, in this case `"external"` is probably good enough, otherwise you could maybe use the `user` or something. (This is not critical info, though)
`last_mc_reg_on` I would set to `serializer.validated_data['authority']`. This is used for some routing in `ussd_public.js`